documentation/add/11_sqw_compression.md

```diff
@@ -0,0 +1,201 @@
+
+The trade-off is between
+- the overhead of recomputing qx,qy,qz,en on-the-fly when reading data, and 
+- the reduced I/O time and the ability to cache entire sqw data on an IDAaaS node (as will be the case of a 200GB sqw file becoming a 50GB file – see explanation for the fourfold reduction in size in a later section).
+
+Preliminary testing by Jacob Wilkins *(results to be collated and confirmed as part of the benchmarking for this epic)* suggest a clear win for the compressed format, notwithstanding the capability gain from caching the entire sqw object.
+
+
+
+### First look at gains
+
+Keeping only the unique information as described above results in the average storage per pixel (assuming float32 for all data) reducing from 36 bytes to (12 + 8f) bytes, where f is the fraction of pixels with non-zero signal (f = 0.01 and 0.13 in the two examples quoted above). The mean storage requirement is therefore reduced to:
+
+**Table 1**
+
+| Fraction of non-zero signal | Bytes per pixel | Percentage of current format
+|-----------------------------|-----------------|-----------------------------
+|  f = 0  (best case) |  12    | 33%
+|  f = 0.011 (RbMnO3) |  12.09 | 34%
+|  f = 0.13 (Fe)      |  13.04 | 36%
+|  f = 0.25           |  14    | 39%
+|  f = 1 (worst case) |  20    | 56%
+		
+For small values of f, the storage is dominated by the space needed to hold {irun, idet, ien} for every pixel. For typical data, the storage requirement is reduced to about 1/3 that of the present sqw files. 
+
+
+
+### The proposed scheme and the achievable compression
+
+To improve on the above, special knowledge of the maximum values of the indices, namely {nrun, ndet, ne} can be used to reduce the space to hold {irun, idet, ien} for every pixel. With the typical values nrun = 500, ndet_max = 70,000, ne_max = 300 then N_max = 1.05e10. If {irun, idet, ien}  is converted into a linear index using the knowledge of nrun, ndet_max, ne_max, then the pixel indexing information can easily be held in a float64 (maximum integer 2^53 = 9e15) i.e. 8 bytes.
+
+In practice, 
+-	With any feasible values for {nrun, ndet_max, ne_max} in the lifetime of ISIS and other facilities, 8 bytes is all that is needed (see additional notes below for details).
+-	Because the overall storage requirement in practice is dominated by {irun,idet,ien}, we could store the signal and error as float64 with little penalty in disk space compared to storing as float32: only for f>0.21 does the penalty exceed 15%. Thie advantage is that the representation in memory and on disk are identical, with little penalty in overall storage in bytes. The following gives the storage needs for the two case (and percentage size compared to the current sqw format):
+
+**Table 2**
+
+| Fraction of non-zero signal | Bytes per pixel float32 signal | Percentage of current format | Bytes per pixel float64 signal | Percentage of current format
+|-----------------------------|-----------------|-----------------------------
```

Comment:
You need additional `|` in this line to match the line above and below, e.g.

```
|---------------------|--------|-----|--------|-----
```

Reply:
Hmm... the .md plugin that Notepad++ uses has no problem parsing to a nice table. I've just seen that the one in Github objects however. I'll make the required change.